vercel/resource_project_environment_variable.go

```diff
@@ -0,0 +1,239 @@
+package vercel
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/provider"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/vercel/terraform-provider-vercel/client"
+)
+
+type resourceProjectEnvironmentVariableType struct{}
+
+// GetSchema returns the schema information for an project environment variable resource.
+func (r resourceProjectEnvironmentVariableType) GetSchema(_ context.Context) (tfsdk.Schema, diag.Diagnostics) {
+	return tfsdk.Schema{
+		Description: `
+Provides a Project Environment Variable resource.
+
+An Project Environment Variable resource allows to add an environment variable to a Vercel project `,
+		Attributes: map[string]tfsdk.Attribute{
+			"target": {
+				Required:    true,
+				Description: "The environments that the environment variable should be present on. Valid targets are either `production`, `preview`, or `development`.",
+				Type: types.SetType{
+					ElemType: types.StringType,
+				},
+			},
+			"key": {
+				Required:      true,
+				PlanModifiers: tfsdk.AttributePlanModifiers{resource.RequiresReplace()},
+				Description:   "The name of the environment variable.",
+				Type:          types.StringType,
+			},
+			"value": {
+				Required:    true,
+				Description: "The value of the environment variable.",
+				Type:        types.StringType,
+			},
+			"git_branch": {
+				Optional:    true,
+				Description: "The git branch of the environment variable",
```

Comment:
We are inconsistent on whether the Description ends with a full-stop or not. Sorry if you've copy pasted that from existing code, but lets make it consistent :)